FirebasePerformance/Sources/AppActivity/FPRAppActivityTracker.m

```diff
@@ -121,6 +150,66 @@ - (FIRTrace *)activeTrace {
   return self.backgroundSessionTrace;
 }
 
+/**
+ * Checks if prewarming is available for the platform on current device.
+ * It is available when running iOS 15 and above.
+ *
+ * @return true if the platform could prewarm apps on the current device
+ */
++ (BOOL)isPrewarmAvailable {
+  if (![[GULAppEnvironmentUtil applePlatform] isEqualToString:@"ios"]) {
+    return NO;
+  }
+  NSString *systemVersion = [GULAppEnvironmentUtil systemVersion];
+  if ([systemVersion length] == 0) {
+    return NO;
+  }
+  return [systemVersion compare:@"15" options:NSNumericSearch] != NSOrderedAscending;
+}
+
+/**
+ RC flag for enabling prewarm-detection using ActivePrewarm environment variable
+ */
+- (BOOL)isActivePrewarmEnabled {
+  PrewarmDetectionMode mode = [self.configurations prewarmDetectionMode];
+  return (mode == OnlyActivePrewarm || mode == EitherActivePrewarmOrDoubleDispatch);
+}
+
+/**
+ RC flag for enabling prewarm-detection using double dispatch method
+ */
+- (BOOL)isDoubleDispatchEnabled {
+  PrewarmDetectionMode mode = [self.configurations prewarmDetectionMode];
+  return (mode == OnlyDoubleDispatch || mode == EitherActivePrewarmOrDoubleDispatch);
+}
+
+/**
+ Checks if the current app start is a prewarmed app start
+ */
+- (BOOL)isApplicationPreWarmed {
+  if (![FPRAppActivityTracker isPrewarmAvailable]) {
+    return NO;
+  }
+
+  // Force drop all events according to RC flag
+  if ([self.configurations prewarmDetectionMode] == DropAllEvents) {
+    return YES;
+  }
```

Comment:
This method should only answer `isApplicationPreWarmed` instead of answering if events should be dropped or not. Can we move this call the caller of `isApplicationPreWarmed`?

Reply:
Moved to a new method `isAppStartEnabled`, called right before `isApplicationPreWarmed`